api/server/router/network/network_routes.go

```diff
@@ -82,14 +84,78 @@ func (n *networkRouter) getNetwork(ctx context.Context, w http.ResponseWriter, r
 		return err
 	}
 
-	nw, err := n.backend.FindNetwork(vars["id"])
-	if err != nil {
-		if nr, err := n.cluster.GetNetwork(vars["id"]); err == nil {
-			return httputils.WriteJSON(w, http.StatusOK, nr)
+	term := vars["id"]
+
+	// In case multiple networks have duplicate names, return error.
+	// TODO (yongtang): should we wrap with version here for backward compatibility?
+
+	// First find based on full ID, return immediately once one is found.
+	// If a network appears both in swarm and local, assume it is in local first
+
+	// For full name and partial ID, same the result first, and process later
+	// in case multiple records was found based on the same term
+	listByFullName := map[string]types.NetworkResource{}
+	listByPartialID := map[string]types.NetworkResource{}
+
+	nw := n.backend.GetNetworks()
+	for _, network := range nw {
+		if network.ID() == term {
+			return httputils.WriteJSON(w, http.StatusOK, *n.buildDetailedNetworkResources(network))
+		}
+		if network.Name() == term {
+			// It may not be necessary to check ID collision (not Name collision) here,
+			// if the ID is guaranteed to be unique in local scope
+			if _, ok := listByFullName[network.ID()]; !ok {
+				listByFullName[network.ID()] = *n.buildDetailedNetworkResources(network)
+			}
+		}
+		if strings.HasPrefix(network.ID(), term) {
+			// It may not be necessary to check ID collision (not Name collision) here,
+			// if the ID is guaranteed to be unique in local scope
+			if _, ok := listByPartialID[network.ID()]; !ok {
+				listByPartialID[network.ID()] = *n.buildDetailedNetworkResources(network)
+			}
+		}
+	}
+
+	nr, _ := n.cluster.GetNetworks()
+	for _, network := range nr {
+		if network.ID == term {
+			return httputils.WriteJSON(w, http.StatusOK, network)
+		}
+		if network.Name == term {
+			if _, ok := listByFullName[network.ID]; !ok {
```

Comment:
Thanks @aboch. The ID check here is probably needed. I noticed that some networks might exist both in local scope and swarm scope. So ID check for the first loop of iterating through local scope is not needed. But in this loop (swarm scope), we may already have the same network from last (local scope).